includes/TagMapper.inc

```diff
@@ -56,25 +76,28 @@ class TagMapper {
 
   /**
    * Biosample attributes.  derived from the "harmonized name" of <Attribute>s
+   *
    * @return array
    */
   private function provideBiosampleAttributeDict() {
 
+    //please keep this alphabetized for sanity.
+    //If updated, please also delete/update terms in the install file.
     return [
-      'age',
-      'bio_material',
-      'breed',
-      'collection_date',
-      'cultivar',
-      'dev_stage',
-      'geo_loc_name',
-      'isolation_source',
-      'orgmod_note',
-      'phenotype',
-      'sex',
-      'strain',
-      'sub_species',
-      'tissue',
+      'age' => tripal_get_cvterm(['id' => 'local:age']),
+      'bio_material' => NULL,
+      'breed' => tripal_get_cvterm(['id' => 'local:breed']),
+      'collection_date' => tripal_get_cvterm(['id' => 'local:collection_date']),
+      'cultivar' => tripal_get_cvterm(['id' => 'local:cultivar']),
+      'dev_stage' => tripal_get_cvterm(['id' => 'local:dev_stage']),
+      'geo_loc_name' => tripal_get_cvterm(['id' => 'local:geo_loc_name']),
+      'isolation_source' => tripal_get_cvterm(['id' => 'local:isolation_source']),
+      'orgmod_note' => tripal_get_cvterm(['id' => 'local:orgmod_note']),
+      'phenotype' => tripal_get_cvterm(['id' => 'local:phenotype']),
+      'sex' => tripal_get_cvterm(['id' => 'local:sex']),
+      'strain' => tripal_get_cvterm(['id' => 'local:strain']),
+      'sub_species' => tripal_get_cvterm(['id' => 'local:sub_species']),
+      'tissue' => tripal_get_cvterm(['id' => 'local:tissue']),
```

Comment:
so this means N queries per instance of this class. Maybe we can bulk load in the constructor into a static cache? If you want I can do that after you merge as an optimization.

Reply:
ok that sounds like a good idea.  I'll make an issue.